apis/core/v1beta1/function_webhook.go

```diff
@@ -167,9 +167,14 @@ func (r *Function) ValidateBuild() error {
 			"must be specified when `spec.build.dockerfile` is not enabled")
 	}
 
-	if r.Spec.Build.SrcRepo != nil && r.Spec.Build.SrcRepo.Url == "" {
-		return field.Required(field.NewPath("spec", "build", "srcRepo", "url"),
-			"must be specified when `spec.build.srcRepo` enabled")
+	if r.Spec.Build.SrcRepo == nil {
+		return field.Required(field.NewPath("spec", "build", "srcRepo"),
+			"must be specified when `spec.build` enabled")
+	}
+
+	if r.Spec.Build.SrcRepo.Url == "" && r.Spec.Build.SrcRepo.BundleContainer == nil {
```

Comment:
Maybe we need to add a check to ensure only one of the URL or bundleContainer can be specified?
Not all of them can be set simultaneously.

Reply:
If `bundleContainer` and `url` are set at the same time, `bundleContainer` will be used first.

Reply:
ok